libnavui-androidauto/src/main/java/com/mapbox/androidauto/internal/car/search/CarFeedbackRepository.kt

```diff
@@ -0,0 +1,33 @@
+package com.mapbox.androidauto.internal.car.search
+
+import com.mapbox.navigation.core.geodeeplink.GeoDeeplink
+import com.mapbox.search.record.FavoriteRecord
+import com.mapbox.search.result.SearchSuggestion
+
+/**
+ * Internal class because [SearchSuggestion] and [FavoriteRecord] will potentially change.
+ *
+ * This object available but likely to change.
+ */
+class CarFeedbackRepository {
+    var feedbackSearchOptions = CarFeedbackSearchOptions()
+        private set
+
+    fun setSearchSuggestions(searchSuggestions: List<SearchSuggestion>) {
+        feedbackSearchOptions = feedbackSearchOptions.copy(
+            searchSuggestions = searchSuggestions
+        )
+    }
+
+    fun setFavoriteRecord(favoriteRecords: List<FavoriteRecord>) {
+        feedbackSearchOptions = feedbackSearchOptions.copy(
+            favoriteRecords = favoriteRecords
+        )
+    }
+
+    fun setGeoDeeplink(geoDeeplink: GeoDeeplink) {
+        feedbackSearchOptions = feedbackSearchOptions.copy(
+            geoDeeplink = geoDeeplink
+        )
+    }
```

Comment:
Do we reset these values anywhere? Looks like when you open the feedback screen again (e.g. in a different scenario), it will reuse the options from the last time.

Reply:
I have been thinking about this.

The current approach relies on easy "screen to screen" communication. You lose this when a screen is defined by a `String`, and it requires an indirect communication with the `MapboxCarContext`. The scope of a screen does not always perfectly match the scope of other state, so compromises here are typical in my opinion.

I've added a `clearFeedbackSearchOptions` function that is called upon arrival. That will reduce the scope to be closer to a single route basis.

Reply:
I think we can do better. These search options are only relevant when the feedback is sent from `PlaceSearchScreen` or `PlacesOnMapScreen`. Therefore I suggest to clear the options when we close any of these screens.

Reply:
> Therefore I suggest to clear the options when we close any of these screens.

Except that will not work because the feedback screen is different from the navigation screen. There are trade-offs

Reply:
An idea, is to make a screen based repository could be based on the string name.

For example, the state lives as long as a screen with "MAPBOX_ACTIVE_GUIDANCE*" is available. That way the state is alive for "MAPBOX_ACTIVE_GUIDANCE" and "MAPBOX_ACTIVE_GUIDANCE_FEEDBACK", but then it is destroyed when "MAPBOX_ARRIVAL" is shown.

Reply:
> Except that will not work because the feedback screen is different from the navigation screen. There are trade-offs

I don't think I understand. Why will it not work? Detecting when `PlaceSearchScreen` or `PlacesOnMapScreen` is closed should be easy.

Reply:
Am going to test this a bit, because it seems that multiple [screens can be started](https://github.com/androidx/androidx/blob/22d0a230b1daccbdabdc9a6658de7e928bea428d/car/app/app/src/main/java/androidx/car/app/ScreenManager.java#L391-L406). 

We could potentially scope the state to screens that are on a backstack

Reply:
> Why will it not work? Detecting when PlaceSearchScreen or PlacesOnMapScreen is closed should be easy.

Ok we can keep state on the backstack, specifically it can be attached to the CREATED state

Reply:
I didn't get to this one yet, but there is a temporary `ScreenLifecycleLog` that will show the lifecycle of the screens on a backstack. So for example, this is the lifecycle for a route request. So you're right that we can scope data to Screen `onCreated/onDestroyed` events.

```
17:14:31.962 ScreenLifecycleLog MAPBOX_FREE_DRIVE onCreate
17:14:32.002 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStart
17:14:32.003 ScreenLifecycleLog MAPBOX_FREE_DRIVE onResume
17:14:35.220 ScreenLifecycleLog MAPBOX_SEARCH onCreate
17:14:35.221 ScreenLifecycleLog MAPBOX_SEARCH onStart
17:14:35.221 ScreenLifecycleLog MAPBOX_FREE_DRIVE onPause
17:14:35.224 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStop
17:14:35.225 ScreenLifecycleLog MAPBOX_SEARCH onResume
17:14:42.108 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onCreate
17:14:42.109 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onStart
17:14:42.109 ScreenLifecycleLog MAPBOX_SEARCH onPause
17:14:42.109 ScreenLifecycleLog MAPBOX_SEARCH onStop
17:14:42.120 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onResume
17:14:44.022 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStart
17:14:44.022 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onPause
17:14:44.025 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onStop
17:14:44.025 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onDestroy
17:14:44.025 ScreenLifecycleLog MAPBOX_SEARCH onDestroy
17:14:44.041 ScreenLifecycleLog MAPBOX_FREE_DRIVE onResume
17:14:44.042 ScreenLifecycleLog MAPBOX_ACTIVE_GUIDANCE onCreate
17:14:44.042 ScreenLifecycleLog MAPBOX_ACTIVE_GUIDANCE onStart
17:14:44.042 ScreenLifecycleLog MAPBOX_FREE_DRIVE onPause
17:14:44.044 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStop
17:14:44.073 ScreenLifecycleLog MAPBOX_ACTIVE_GUIDANCE onResume
17:14:44.073 ScreenLifecycleLog MAPBOX_FREE_DRIVE onDestroy
```

Reply:
Ok I worked on this a bit and it still makes the most sense to clear the feedback data during the arrival screen.

This is feedback data that will help us anytime someone gives feedback, this includes the active guidance state.

```
10:40:05.839 ScreenLifecycleLog MAPBOX_FREE_DRIVE onCreate
10:40:05.888 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStart
10:40:05.891 ScreenLifecycleLog MAPBOX_FREE_DRIVE onResume
10:40:14.082 ScreenLifecycleLog MAPBOX_SEARCH onCreate
10:40:14.083 ScreenLifecycleLog MAPBOX_SEARCH onStart
10:40:14.083 ScreenLifecycleLog MAPBOX_FREE_DRIVE onPause
10:40:14.086 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStop
10:40:14.086 ScreenLifecycleLog MAPBOX_SEARCH onResume
10:40:24.788 ScreenLifecycleLog setSearchSuggestions 5
10:40:25.437 ScreenLifecycleLog setSearchSuggestions 5
10:40:28.669 ScreenLifecycleLog setSearchSuggestions 5
10:40:32.641 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onCreate
10:40:32.642 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onStart
10:40:32.642 ScreenLifecycleLog MAPBOX_SEARCH onPause
10:40:32.642 ScreenLifecycleLog MAPBOX_SEARCH onStop
10:40:32.652 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onResume
10:40:44.827 ScreenLifecycleLog MAPBOX_SEARCH onStart
10:40:44.828 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onPause
10:40:44.834 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onStop
10:40:44.834 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onDestroy
10:40:44.835 ScreenLifecycleLog MAPBOX_SEARCH onResume
10:40:49.565 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onCreate
10:40:49.566 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onStart
10:40:49.566 ScreenLifecycleLog MAPBOX_SEARCH onPause
10:40:49.566 ScreenLifecycleLog MAPBOX_SEARCH onStop
10:40:49.567 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onResume
10:40:53.342 ScreenLifecycleLog MAPBOX_FREE_DRIVE onStart
10:40:53.343 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onPause
10:40:53.346 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onStop
10:40:53.346 ScreenLifecycleLog MAPBOX_ROUTE_PREVIEW onDestroy
10:40:53.347 ScreenLifecycleLog MAPBOX_SEARCH onDestroy
10:40:53.347 ScreenLifecycleLog setSearchSuggestions null <<< === Notice the data is gone when active guidance starts 
10:40:53.367 ScreenLifecycleLog MAPBOX_FREE_DRIVE onResume
10:40:53.368 ScreenLifecycleLog MAPBOX_ACTIVE_GUIDANCE onCreate
10:40:53.369 ScreenLifecycleLog MAPBOX_ACTIVE_GUIDANCE onStart
```

Reply:
So this thread we discovered a good way to scope data with Android Auto `Screen` lifecycles. But CoPilot is the best solution for capturing the search data here, so it is now deleted from the aa-sdk

the CarRoutePreviewRequest scopes its own data in the way we discussed. It has a repository, and the data is scoped to the screen that uses it. `PlacesListOnMapScreen` or `PlaceSearchScreen`. Only one screen can be used at a time in the `Screen` stack.